test/e2e/atlas/kubernetes_config_generate_test.go

```diff
@@ -18,22 +18,147 @@ package atlas_test
+		assert.Eventually(kh.t, func() bool {
+			_, err := kh.atlasClient.Projects.Delete(context.Background(), kh.project.ID)
+			if !assert.NoError(kh.t, err, "can not delete test project", kh.project.ID) {
+				return false
+			}
+			_, _, err = kh.atlasClient.Projects.GetOneProject(context.Background(), kh.project.ID)
+			if err != nil {
+				var apiError *atlas.ErrorResponse
+				if errors.As(err, &apiError) && (apiError.ErrorCode == "GROUP_NOT_FOUND" || apiError.ErrorCode == ResourceNotFound) {
+					return true
+				}
+				return false
+			}
+			return true
+		}, defaultTimeout, defaultTick, "project should've been deleted", project.ID)
+	})
+}
+
+func (kh *K8SHelper) NewCluster(cluster *atlas.Cluster) {
+	kh.t.Helper()
+
+	require.True(kh.t, kh.project != nil, "can not create cluster without a project")
+	createdCluster, _, err := kh.atlasClient.Clusters.Create(context.Background(), kh.project.ID, cluster)
+	require.Nil(kh.t, err, "error while creating cluster")
+
+	kh.clusters[cluster.Name] = createdCluster
+
+	kh.t.Cleanup(func() {
+		assert.Eventually(kh.t, func() bool {
+			_, err := kh.atlasClient.Clusters.Delete(context.Background(), kh.project.ID, cluster.Name)
+			if !assert.NoError(kh.t, err, "can not delete test cluster", cluster.Name) {
+				return false
+			}
+			_, _, err = kh.atlasClient.Clusters.Get(context.Background(), kh.project.ID, cluster.Name)
+			if err != nil {
+				var apiError *atlas.ErrorResponse
+				if errors.As(err, &apiError) && (apiError.ErrorCode == "CLUSTER_NOT_FOUND" || apiError.ErrorCode == ResourceNotFound) {
+					return true
+				}
+				return false
+			}
+			return true
+		}, defaultTimeout, defaultTick, "cluster should've been deleted", cluster.Name, cluster.ID)
+	})
+}
+
+func (kh *K8SHelper) NewServerlessInstance(instance *atlas.ServerlessCreateRequestParams) {
+	kh.t.Helper()
+	require.True(kh.t, kh.project != nil, "can not create serverless instance without a project")
+	createdInstance, _, err := kh.atlasClient.ServerlessInstances.Create(context.Background(), kh.project.ID, instance)
+	require.Nil(kh.t, err, "error while creating serverless instance")
+
+	kh.clusters[createdInstance.Name] = createdInstance
+	kh.t.Cleanup(func() {
+		assert.Eventually(kh.t, func() bool {
+			_, err := kh.atlasClient.ServerlessInstances.Delete(context.Background(), kh.project.ID, createdInstance.Name)
+			if !assert.NoError(kh.t, err, "can not delete test cluster", createdInstance.Name) {
+				return false
+			}
+			_, _, err = kh.atlasClient.ServerlessInstances.Get(context.Background(), kh.project.ID, createdInstance.Name)
+			if err != nil {
+				var apiError *atlas.ErrorResponse
+				if errors.As(err, &apiError) && (apiError.ErrorCode == "SERVERLESS_INSTANCE_NOT_FOUND" || apiError.ErrorCode == ResourceNotFound) {
+					return true
+				}
+				return false
+			}
+			return true
+		}, defaultTimeout, defaultTick, "serverless instance should've been deleted", createdInstance.Name, createdInstance.ID)
+	})
+}
```

Comment:
can you move this to `helper_test.go`?

Reply:
Do you want to use it in other tests? Or is it just for readability?

Reply:
it would be great to move all utils functions to the `helper_test.go` so that it can also be used by other tests.

Reply:
Ok, I moved this helper to the `helper_test.go`